internal/awslambda/extension/telemetry/server_test.go

```diff
@@ -0,0 +1,111 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewServer(t *testing.T) {
+	t.Parallel()
+
+	assert.NotNil(t, NewServer(logrus.New(), NoopHook()))
+}
+
+func TestServerStart(t *testing.T) {
+	t.Parallel()
+
+	s := NewServer(logrus.New(), NoopHook(), WithCustomAddr("127.0.0.1:8083"))
```

Comment:
Avoid using dedicated ports since if other tests or services on the machine running could clash causing a flake.